capstone/src/main/java/com/google/sps/servlets/EditProfileServlet.java

```diff
@@ -0,0 +1,76 @@
+package com.google.sps.servlets;
+
+import com.google.sps.Objects.Badge;
+import com.google.sps.Objects.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import error.ErrorHandler;
+
+@WebServlet("/editProfile")
+public class EditProfileServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String first = request.getParameter("first");
+    String last = request.getParameter("last");
+    String email = request.getParameter("email");
+    String phone = request.getParameter("phone");
+    ArrayList<String> interests = getInterests(request);
+    
+    UserService userService = UserServiceFactory.getUserService();
+    String userId = userService.getCurrentUser().getUserId();
+    // not error checking because this class will extend AuthenticatedServlet 
+    // (will implement once Lucy's PR with the class is merged to master)
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Key entityKey = KeyFactory.createKey("User", userId);
+    Entity userEntity;
+    try {
+      userEntity = datastore.get(entityKey);
+    } catch (EntityNotFoundException e) {
+      ErrorHandler.sendError(response, "User not logged in.");
+      return;
+    }
+    
+    User user = User.fromEntity(userEntity);
+    user.setFirstName(first);
+    user.setLastName(last);
+    user.setEmail(email);
+    user.setPhoneNumber(phone);
+    user.setInterests(interests);
+
+    datastore.put(user.toEntity());
```

Comment:
This is a broad comment I made in a 1:1 with Lucy, but I'll try to reiterate for everyone: @anikabagga @lxcyqx 

Due to how you have to structure your servlets to match a very specific interface, the only way I can figure out what each servlet does is:
1. Read the name of the servlet, to guess what it's supposed to do on a Post/Get
2. Look all the lines in 20-40 line function called doPost, and read over all the parts of it to figure out what data it's getting in the request and what the outcome of that request is.

I think to minimize the cognitive load around figuring out what each servlet does, I would like to request any action that's being taken by the server exist within a well-named function. This doesn't mean just move all this logic into a helper function and call it, leaving doPost to be a one line function, but I would say that doPost can get a request, format data however it needs to be formatted, do any logic tests/checking/validation that you have a valid post, and once you've done those things, consider calling a well-named helper function.

In this case, a `doPost` on `editProfileServlet` should likely call a helper function called `updateProfile`, so that I can easily read within just those 3 names, what servlet it is, is it a get or a post, and ultimately what action the server is making in response.